Core/include/Acts/Propagator/detail/SteppingHelper.hpp

```diff
@@ -33,17 +33,18 @@ namespace detail {
 template <typename stepper_t>
 Acts::Intersection3D::Status updateSingleSurfaceStatus(
     const stepper_t& stepper, typename stepper_t::State& state,
-    const Surface& surface, Direction navDir, const BoundaryCheck& bcheck,
-    ActsScalar surfaceTolerance, const Logger& logger) {
+    const Surface& surface, std::uint8_t index, Direction navDir,
+    const BoundaryCheck& bcheck, ActsScalar surfaceTolerance,
+    const Logger& logger) {
   ACTS_VERBOSE(
       "Update single surface status for surface: " << surface.geometryId());
 
   auto sIntersection = surface.intersect(
       state.geoContext, stepper.position(state),
-      navDir * stepper.direction(state), bcheck, surfaceTolerance);
+      navDir * stepper.direction(state), bcheck, surfaceTolerance)[index];
```

Comment:
So here, in theory, we'd have the intersection already at the call-site, right? So maybe it would really make sense to add the intersection as an argument explicitly.

Reply:
I can try how badly this affects the calling code